tests/api/api/test_functions.py

```diff
@@ -481,6 +481,13 @@ def test_tracking_on_serving(
     Validate that `.set_tracking()` configurations are applied to
     a serving function for model monitoring.
     """
+    server.api.utils.singletons.k8s.get_k8s_helper().v1api = unittest.mock.Mock()
+
+    config_map = unittest.mock.Mock()
+    config_map.items = []
+    server.api.utils.singletons.k8s.get_k8s_helper().v1api.list_namespaced_config_map = unittest.mock.Mock(
+        return_value=config_map
+    )
```

Comment:
You need to revert these changes at the end of the test. Since you have `monkeypatch` available, best to use it to perform these patching, as it will take care of reverting automatically.